src/Components/AddIssue/index.tsx

```diff
@@ -2,138 +2,180 @@ import React, { ChangeEvent } from 'react';
-    }
-
-    handleChangeOfIssueTypeDropDownList=(event:React.ChangeEvent<HTMLSelectElement>): void => {
-      const index = Number(event.target.value);
-      let issueType = this.state.issueTypes[index];
-      let issue = this.state.issue;
-      issue.IssueTypeAndKey.IssueTypeAndId = issueType;
-      this.setState({issue:issue});
-
-      this.props.onChangeIssue(issue, this.props.index);
-    }
-
-    handleChangeOfIssueLocationDropDownList=(event:React.ChangeEvent<HTMLSelectElement>): void => {
-      console.log("handleChangeOfIssueLocationDropDownList");
-      
-      const index = Number(event.target.value);
-      let location = this.state.issueLocations[index];
-
-      let issue = this.state.issue;
-      issue.Location = location;
-      this.setState({issue:issue });
-
-      this.props.onChangeIssue(issue, this.props.index);
-    }
-
-    renderIssueType(issueType: IIssueTypeAndId, index: number) {
-      return (
-        <option key={index} value={index}>{issueType.IssueType}</option>
-      );
-    }
-
-    renderLocation(location: IIssueLocation, index:number)
-    {
-      return(
-        <option key={location.key} value={index}>{location.name}</option>
-      );
-    }
-
-    renderFirstOption(text: string){
-      return (<option value="" disabled hidden>{text}</option>)
-    }
-
-
-    render() {
-      return (
-        <div>
-          <div>
-            <label data-test="issue-label" className="label">Issue Type</label>
-            <br/>
-            <select id="issue-dropdown" data-test="issue-dropdown" className="select" onChange={this.handleChangeOfIssueTypeDropDownList} name="IssueType" value={this.state.issue.IssueTypeAndKey.IssueTypeAndId.IssueType}>
-              {this.renderFirstOption("Select Issue Type")}
-              {this._issueTypes.map(this.renderIssueType)}
-            </select>
-          </div>
-          <div>
-            <label data-test="location-label" className="label">Location of issue</label>
-            <br />
-            <select id="location-dropdown" data-test="location-dropdown" className="select" onChange={this.handleChangeOfIssueLocationDropDownList} name="IssueLocation">
-              {this.renderFirstOption("Select Location")}
-              {this._issueLocations.map(this.renderLocation)}
-            </select>
-          </div>
-          <div>
-          <div className="label">Notes about issue</div>
-          <textarea className="note-input-box" data-test="notes" id="issue-note" value={this.state.issue.Notes} onChange={this.handleChangeOfIssueNote } name= "notes"/>
-          </div>
-          <button id="delete-issue" className="button btn-primary btn-stacked" data-test="delete-issue" onClick={this.deleteIssue}>Delete Issue</button>
+export class AddIssue extends React.Component<IAddIssuesProps, IAddIssueState> {
+  public static contextType = ServiceContext;
+
+  private _issueTypes = Array.from<IssueType>(issueTypesData);
```

Comment:
Not sure how I feel about underscore syntax in typescript @PinkhamSoftware but don't care that much 😄 

I think the fact that you have to explicitly write `this.thing` makes calling the variable `_thing` unnecessary.